c10/util/flat_hash_map.h

```diff
@@ -234,12 +234,14 @@ template <
     typename T,
     typename FindKey,
     typename ArgumentHash,
-    typename Hasher,
+    typename DetailHasher,
```

Comment:
Is this change necessary? Seems that it's simply changing the name of `Hasher` to `DetailHasher`. I believe this file was copied over from [this repo](https://github.com/skarupke/flat_hash_map) so I think we should avoid making changes unless necessary.

Reply:
It's mentioned in PR description, isn't it
> Rename Hasher typename since it's not compiling due to https://developercommunity.visualstudio.com/t/1397858